homeassistant/components/wiffi/binary_sensor.py

```diff
@@ -0,0 +1,54 @@
+"""Binary sensor platform support for wiffi devices."""
+
+import logging
+from pathlib import Path
+
+from homeassistant.const import STATE_OFF, STATE_ON
+
+from .const import DOMAIN
+from .entity_base import WiffiEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up platform for a new integration.
+
+    Called by the HA framework after async_forward_entry_setup has been called
+    during initialization of a new integration (= wiffi).
+    """
+    stem = Path(__file__).stem  # stem = filename without py
+    hass.data[DOMAIN][config_entry.entry_id].async_add_entities[
+        stem
+    ] = async_add_entities
+
+
+class BoolEntity(WiffiEntity):
```

Comment:
We must inherit BinarySensorDevice. That's the home assistant design.
```suggestion
class BoolEntity(WiffiEntity, BinarySensorDevice):
```

Reply:
done